### PR TITLE
MIPD: use EIP-6963 to discover injected EVM providers

### PR DESCRIPTION
**Pull request by tanguyenvn**
image
image

### CODE REVIEW

File: packages/adapters/default-evm-adapter/src/injectedEvmAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+  async disconnect(options: { cleanup: boolean } = { cleanup: false }): Promise<void> {
+    await super.disconnectSession();
+    if (typeof this.injectedProvider?.removeAllListeners !== "undefined") this.injectedProvider?.removeAllListeners();
+    try {
+      await this.injectedProvider.request({
+        method: "wallet_revokePermissions",
+        params: [{ eth_accounts: {} }],
+      });
+    } catch (error) {}
```
Comment (tanguyenvn) [inline packages/adapters/default-evm-adapter/src/injectedEvmAdapter.ts:102]: use this method to disconnect from Metamask, may not work in some wallets

File: packages/modal/src/modalManager.ts [hunk 2]
```diff
@@ -108,7 +116,7 @@ export class Web3Auth extends Web3AuthNoModal implements IWeb3AuthModal {
     const adapterConfigurationPromises = allAdapters.map(async (adapterName) => {
       // start with the default config of adapter.
       let adapterConfig = this.modalConfig.adapters?.[adapterName] || {
-        label: adapterName,
+        label: adapterName.split("-").map(capitalizeFirstLetter).join(" "),
```
Comment (tanguyenvn) [inline packages/modal/src/modalManager.ts:119]: magic-eden adapter -> show label Magic Eden

File: packages/ui/src/components/ExternalWallets.tsx [hunk 3]
```diff
@@ -204,7 +204,7 @@ export default function ExternalWallet(props: ExternalWalletsProps) {
               const providerIcon = button.isLink ? (
                 <img src={button.logo} height="auto" width="auto" alt={`login-${button.name}`} />
               ) : (
-                <Image imageId={`login-${button.name}`} hoverImageId={`login-${button.name}`} isButton />
+                <Image imageId={`login-${button.name}`} hoverImageId={`login-${button.name}`} height="32" width="32" isButton />
```
Comment (tanguyenvn) [inline packages/ui/src/components/ExternalWallets.tsx:207]: some SVGs have larger heights and widths

Comment (chaitanyapotti) [reply packages/ui/src/components/ExternalWallets.tsx:207]: Ack

File: packages/modal/src/modalManager.ts (hunk 2 above)
Comment (chaitanyapotti) [reply packages/modal/src/modalManager.ts:119]: Good

File: packages/adapters/base-evm-adapter/src/baseEvmAdapter.ts [hunk 4]
```diff
@@ -45,10 +45,11 @@ export abstract class BaseEvmAdapter<T> extends BaseAdapter<T> {
       };
 
       const challenge = await signChallenge(payload, chainNamespace);
+      const hexChallenge = `0x${Buffer.from(challenge, "utf8").toString("hex")}`;
```
[the remark this answers is not in the document]
Comment (chaitanyapotti) [reply packages/adapters/base-evm-adapter/src/baseEvmAdapter.ts:48]: Ok. seems like the api changed along the way

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**